cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaMetadataStateManagerMockTest.java

```diff
@@ -0,0 +1,640 @@
+            .endMetadata()
+            .withNewSpec()
+                .withNewKafka()
+                .withListeners(new GenericKafkaListenerBuilder()
+                        .withName("plain")
+                        .withPort(9092)
+                        .withType(KafkaListenerType.INTERNAL)
+                        .withTls(false)
+                        .build())
+            .endKafka()
+            .endSpec()
+            .withNewStatus()
+                .withKafkaMetadataState(ZooKeeper.name())
+            .endStatus()
+            .build();
+
+    private static KafkaCluster createKafkaCluster(Kafka kafka)   {
+        return KafkaCluster.fromCrd(
+                Reconciliation.DUMMY_RECONCILIATION,
+                kafka,
+                POOLS,
+                VERSIONS,
+                KafkaVersionTestUtils.DEFAULT_ZOOKEEPER_VERSION_CHANGE,
+                KafkaMetadataConfigurationState.ZK,
+                null,
+                SHARED_ENV_PROVIDER);
+    }
+
+    private final static KafkaNodePool CONTROLLERS = new KafkaNodePoolBuilder()
+            .withNewMetadata()
+                .withName("controllers")
+                .withNamespace(NAMESPACE)
+                .withGeneration(1L)
+                .withLabels(Map.of(Labels.STRIMZI_CLUSTER_LABEL, CLUSTER_NAME))
+            .endMetadata()
+            .withNewSpec()
+                .withReplicas(3)
+                .withNewJbodStorage()
+                    .withVolumes(new PersistentClaimStorageBuilder().withId(0).withSize("100Gi").build())
+                .endJbodStorage()
+                .withRoles(ProcessRoles.CONTROLLER)
+                .withResources(new ResourceRequirementsBuilder().withRequests(Map.of("cpu", new Quantity("4"))).build())
+            .endSpec()
+            .build();
+    private final static KafkaNodePool BROKERS = new KafkaNodePoolBuilder()
+            .withNewMetadata()
+                .withName("brokers")
+                .withNamespace(NAMESPACE)
+                .withGeneration(1L)
+                .withLabels(Map.of(Labels.STRIMZI_CLUSTER_LABEL, CLUSTER_NAME))
+            .endMetadata()
+            .withNewSpec()
+                .withReplicas(3)
+                .withNewJbodStorage()
+                    .withVolumes(new PersistentClaimStorageBuilder().withId(0).withSize("200Gi").build())
+                .endJbodStorage()
+                .withRoles(ProcessRoles.BROKER)
+                .withResources(new ResourceRequirementsBuilder().withRequests(Map.of("cpu", new Quantity("6"))).build())
+            .endSpec()
+            .build();
+    private static final List<KafkaPool> POOLS = NodePoolUtils.createKafkaPools(Reconciliation.DUMMY_RECONCILIATION, KAFKA, List.of(CONTROLLERS, BROKERS), Map.of(), Map.of(), true, SHARED_ENV_PROVIDER);
+
+    /** Tests migration from Zookeeper State to KRaftPostMigration State.
```

Comment:
start the comment on the next line. Please do the same for all the other Javadocs.

```suggestion
    /**
    * Tests migration from Zookeeper State to KRaftPostMigration State.
```